CHANGES.md

```diff
@@ -1,6 +1,361 @@
+  (see TODO)
+* eliminated unnecessary synchronization for reductions on AMD GPUs
+
+Performance Optimizations / Improvements
+----------------------------------------
+* significantly improved `Sort.sort()` performance for large problem sizes
+* eliminated extraneous array copies in initializers caused by domain queries
+* improved codegen for `sqrt()` and `abs()` to directly map to LLVM intrinsics
+* adjusted the LLVM optimizer to assume math functions don't set C's `errno`
+
+Improvements to Compilation Times / Generated Code
+--------------------------------------------------
+* reduced `chpl` memory footprint for large programs via the compiler driver  
+  (see https://chapel-lang.org/docs/2.0/technotes/driver.html)
+
+Memory Improvements
+-------------------
+
+Tool Improvements
+-----------------
+* added `chpl-language-server`, a language server implementation for Chapel  
+  (see https://chapel-lang.org/docs/2.0/tools/chpl-language-server/chpl-language-server.html)
+* added support for user-defined lint rules in `chplcheck`  
+  (see https://chapel-lang.org/docs/2.0/tools/chplcheck/chplcheck.html)
+* improved how locations are reported by tools making use of the `chpl` parser
+* improved rendering of `chplcheck` lint rules in editors
+* improved rendering of return intents by `chpldoc` for some browsers
+
+Documentation Improvements
+--------------------------
+* merged 'ChapelIO' documentation into the 'IO' module's documentation  
+  (see https://chapel-lang.org/docs/2.0/modules/standard/IO.html)
+* fixed a browser-specific rendering issue with spacing of return intents
+* updated platform documentation for AWS to reflect current best practices  
+  (see https://chapel-lang.org/docs/2.0/platforms/aws.html)
+* improved InfiniBand documentation for clarity and to mention `pmi` launches
+  (see https://chapel-lang.org/docs/2.0/platforms/infiniband.html)
+* refreshed documentation concerning troubleshooting GASNet runs  
+  (see https://chapel-lang.org/docs/2.0/usingchapel/multilocale.html#troubleshooting)
+* clarified required conditions for split initialization  
+  (see https://chapel-lang.org/docs/2.0/language/spec/variables.html#split-initialization))
+* clarified required conditions for copy elision  
+  (see https://chapel-lang.org/docs/2.0/language/spec/variables.html#copy-elision)
+* improved test system documentation to cover more features
+  (see TODO)
+* improved documentation of enum constants to leverage new 'chpldoc' features
+* fixed typos in the `Math.sqrt()` documentation
+* fixed some cases where documentation and argument lists did not match
+* fixed a bug in the spec where certain statement forms lacked semicolons
+* fixed various typos and formatting issues
+
+Example Codes
+-------------
+
+Syntax Highlighting
+-------------------
+
+Configuration / Build / Packaging Changes
+-----------------------------------------
+* updated Chapel prereqs to require CMake 3.20, C++ 17, and those of LLVM 17  
```

Comment:
"and those of LLVM 17" reads oddly to me

What about

> * updated Chapel prereqs to match LLVM 17's requirements
>  (CMake 3.20, C++ 17, GCC 7.4)

Reply:
Oh, were all of the prereq updates due to LLVM 17?  (this was me merging items from two different developers which is how we got to this state).

Reply:
yes, those updates to cmake and c++17 are both due to LLVM version bump

Reply:
Chapel doesn't require gcc 7.4, though, right?  (i.e., if I'm on a system with no gcc, I'm not in trouble)  It's more like "If using gcc, we require gcc 7.4 or later?"  Is that redundant with saying "C++ 17" or are there other things necessitating those versions of gcc?

Reply:
That's correct. It's hard to say it in a short way; we don't require GCC but if you are using GCC it needs to be GCC 7.4.

Reply:
LLVM requires gcc 7.4 or later to build (or an equivalent clang/other compiler). I think earlier gcc versions do fully support C++17.

Reply:
Thanks, I think just moving this into a bullet of its own is the easiest way to deal with this.  I.e., I don't see it as being particularly LLVM-specific.  Or, we could also just avoid saying it at all since it's hard to quantify what "equivalent clang/other compiler" would be.  But maybe gcc is common enough to deserve a line of its own.